plugin/completion.py

```diff
@@ -147,12 +147,34 @@ def on_query_completions(self, prefix: str, locations: List[int]) -> Optional[su
         if not self.enabled:
             return None
 
+        prefix = self.include_special_chars(prefix, locations[0])
         completion_list = sublime.CompletionList()
 
         self.do_request(completion_list, prefix, locations)
 
         return completion_list
 
+    def include_special_chars(self, prefix: str, point: int) -> str:
+        """
+        Sanatize the prefix, sublime trims of special characters from the prefix
+        """
+        special_chars = ['$']
```

Comment:
I'm guessing the special chars are exactly the `view.settings().get("word_separators")`

Reply:
Perhaps you can test this by removing `$` from the word separators and then testing if sublime still trims `$`.

Reply:
If `$` is troublesome for PHP, I'm also predicting that Powershell will give us troubles. Prefixes for Powershell are `-`.

Reply:
I thought that word_separators are the cause of this. But they are not.
I removed the $ from the list. But I experience the same problem.
"word_separators": "./\\()\"'-:,.;<>~!@#%^&*|+=[]{}`~?",

I don't understand why sublime removes the $ when selecting a command completion item in the first place. Because the $ is by default a word separator. And by my logic it should not be removed when selecting a command completion because it is not part of the word.

Reply:
Consider trying out Powershell with PowershellEditorServices.

Try the syntax from the master branch at https://github.com/SublimeText/PowerShell
Then try the syntax from the ongoing rewrite: https://github.com/SublimeText/PowerShell/pull/165

Notably, the PR author decided to remove `-` from the word separators. I'm wondering out loud what that would entail for our logic.

Reply:
> I thought that word_separators are the cause of this. But they are not. I removed the $ from the list. But I experience the same problem.

Bummer. I had hoped we would find the solution through word_separators :)

Reply:
I will try to setup the PowershellEditorServices. 
What would you like me to test exactly? :)

Reply:
> What would you like me to test exactly?

Create a test file test.ps1 and start writing `get-` or `$env:`, completions should appear and e.g. `-` symbols should not disappear, and `$` symbols should also not disappear.

Reply:
I have no luck setting up PowershellEditorServices on linux right now :( 
Do you maybe have it setup?